narwhals/_expression.py

```diff
@@ -28,9 +32,13 @@ def extract_native(expr: Expr, other: Any) -> Any:
 
 
 class Expr:
-    def __init__(self, call: Callable[[Any], Any]) -> None:
+    def __init__(self, call: Callable[[Any], Any], api_version: API_VERSION) -> None:
         # callable from namespace to expr
         self._call = call
+        self._api_version = api_version
+
+    def _from_expr(self, call: Callable[[Any], Any]) -> Self:
```

Comment:
what do you think about changing the name to something like
```suggestion
    def _from_native_expr(self, call: Callable[[Any], Any]) -> Self:
```

Reply:
nice!